Лабораторная работа 2

WildWestMouse opened the pull request.

WildWestMouse committed:
Лабораторная работа 2

rmushkin commented on Лабораторная работа 2/task2.py:
```diff
@@ -0,0 +1,11 @@
+increase = 0.03  # Ежемесячный рост цен
+money_capital = 0
+
+for month in range(months,0,-1):
```
1. PEP8
2. `month` нигде не используется, поэтому питоничнее использовать `_` вместо имени переменной. Это полностью валидное имя, т. к. нижнее подчёркивание считается обычной буквой; однако выглядит как "пустое место" и дополнительно самодокументирует код, указывая на неиспользуемость значения.
3. По поскольку вы не исподльзуете `month`, то и направление вообще не важно (кстати, я вообще не очень понимаю, почему вам захотелось идти от большего к меньшему)

rmushkin commented on Лабораторная работа 2/task2.py:
```diff
@@ -0,0 +1,11 @@
+    money_capital += spend - salary
+    spend += spend * increase
+
+print(f"Подушка безопасности, чтобы протянуть {months} месяцев без долгов:", round(money_capital))
```
Ловушка с `round`: 3 рублей не хватит, чтобы купить гвоздь за 3 рубля 25 копеек. Надо округлять в большую сторону.

rmushkin commented on Лабораторная работа 2/task1.py:
Представьте, что `spend` изначально больше, чем `money_capital + salary`. В такой ситуации вы не продержитесь и месяца, не уйдя в минус... Но ваш код выведет "1". Это неправильно.

rmushkin reviewed (changes requested):
Пожалуйста, поправьте замечания! Если что, пишите.

WildWestMouse committed:
Исправление ошибок

Подправил task1: Теперь, при spend > money_capital + salary, мы получаем 0, как и должно быть.
В task2 изменил неиспользуемую переменную month и округлил результат в большую сторону.

rmushkin reviewed (approved):
Принимается ✅
